redis/src/cluster.rs

```diff
@@ -708,10 +705,17 @@ where
                         conn.req_packed_command(&b"*1\r\n$6\r\nASKING\r\n"[..])?;
                     }
                     (addr.to_string(), conn)
-                } else if route.is_none() {
-                    get_random_connection(&mut connections)
                 } else {
-                    self.get_connection(&mut connections, route.as_ref().unwrap())?
+                    match &single_node_routing {
+                        SingleNodeRoutingInfo::Random => get_random_connection(&mut connections),
+                        SingleNodeRoutingInfo::SpecificNode(route) => {
+                            self.get_connection(&mut connections, route)?
```

Comment:
so get_connection returns the identifier, which doesn't have to be in equal to address in the future, however for SingleNodeRoutingInfo::Random and SingleNodeRoutingInfo::ByAddress returning the address won't change.
this might cause a later issue

Reply:
I don't understand your comment - this is the sync client. there's no identifier here.